core/tests/block.rs

```diff
@@ -784,3 +784,131 @@ fn validate_header_proof() {
 	)
 	.is_err());
 }
+
+// Test coverage for verifying cut-through during block validation.
+// It is not valid for a block to spend an output and produce a new output with the same commitment.
+// This test covers the case where a plain output is spent, producing a plain output with the same commitment.
+#[test]
```

Comment:
This test seems to duplicate the test in tests/transaction.rs ?!

Reply:
Starts off the same but tests block validity by building a block from the transaction.

Reply:
ah, yes. would it make sense to avoid the redundancy by having both tests calling a single helper function to build this tx?